### PR TITLE
New recipe for library jsoncons

### PR DESCRIPTION
**Pull request by visze**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by visze** [comment]
@conda-forge/help-c-cpp how can I test that a library can be included

**Comment by scopatz** [comment]
@visze - what do you mean exactly?

**Comment by visze** [comment]
@scopatz I need tests otherwise linting failed. This library is a siimple copy into the include folder (see `build.sh`). Just looking for the files like:

```yml
test:
  files:
    - include/jsoncons/source.hpp
    - include/jsoncons_ext/csv/csv.hpp
```

Is not enough for linting.

I tried using the commands option in test. This is fine for lint. but it failes. This is what I use:

```yml
test:
  commands:
    - test -f $PREFIX/include/jsoncons/source.hpp
```

Any suggestions how to test this library?

File: recipes/jsoncons/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+  run:
+
+test:
+  files:
+    - include/jsoncons/source.hpp
+    - include/jsoncons_ext/csv/csv.hpp
```
Comment (scopatz) [inline recipes/jsoncons/meta.yaml:25]: These should be installed

File: recipes/jsoncons/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+build:
+  skip: true  # [win]
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+  run:
```
Comment (scopatz) [inline recipes/jsoncons/meta.yaml:20]: Please remove empty entries

File: recipes/jsoncons/build.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+cp -rv include/jsoncons $PREFIX/include/
```
Comment (scopatz) [inline recipes/jsoncons/build.sh:1]: I think you need a `mkdir -p $PREFIX/include/`

File: recipes/jsoncons/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    - include/jsoncons_ext/csv/csv.hpp
+  commands:
+    - test -f $PREFIX/include/jsoncons/source.hpp 
+
+about:
+  home: https://github.com/danielaparker/jsoncons
+  license: BSL-1.0
+  license_family: OTHER
+  license_file: LICENSE
+  summary: ' header-only library for constructing JSON and JSON-like data formats such as CBOR.'
```
Comment (scopatz) [inline recipes/jsoncons/meta.yaml:34]: No space at start

**Pull request closed without merging**